internal/telemetry/service_test.go

```diff
@@ -2,14 +2,49 @@ package telemetry
 
 import (
 	"errors"
+	"io/ioutil"
+	"log"
+	"os"
 	"testing"
 
 	"github.com/10gen/realm-cli/internal/utils/test/assert"
 )
 
 func TestNewService(t *testing.T) {
 	t.Run("Should create the expected Service", func(t *testing.T) {
-		service := NewService(ModeOn, "userID", "command")
+		service := NewService(ModeStdout, "userID", nil, "command")
+
+		assert.Equal(t, "command", service.command)
+		assert.True(t, service.executionID != "", "service execution id must not be blank")
+		assert.Equal(t, "userID", service.userID)
+		assert.NotNil(t, service.tracker)
+	})
+
+	createSegmentFn := func() *Service {
+		return NewService(ModeOn, "userID", log.New(os.Stdout, "LogPrefix ", log.Lmsgprefix), "command")
+	}
+	t.Run("Should create a segment tracking service if the segmentWriteKey is there", func(t *testing.T) {
+		swk := segmentWriteKey
+		defer func() { segmentWriteKey = swk }()
+
+		segmentWriteKey = "testing"
+		testServiceStdoutput(t, createSegmentFn, "")
+
+		service := createSegmentFn()
+
+		assert.Equal(t, "command", service.command)
```

Comment:
these asserts are already verified with `ModeStdout` ... we don't need them here as well (since the logic that handles setting these details is independent of whichever `mode` is passed in)